### PR TITLE
feat: integrate Sift SDK (PRO-87)

### PR DESCRIPTION
**Pull request by mdole**
This PR resolves [PRO-87]

### Description

Uses [Sift's RN wrapper](https://github.com/SiftScience/sift-react-native) to send "device information and app life cycle events" to Sift ([docs](https://sift.com/developers/docs/curl/javascript-api/overview)).

Noting that I've already updated the .env.shared that lives in S3, as well as CircleCI env, with the new variables.

This is unfortunately pretty much black-box; I can see what payloads show up in our Sift sandbox but there doesn't appear to be any other way to get at what's triggered when etc.

Event payloads look like:

```json
{
  "data": [
    {
      "user_id": "5744b8888b3b8145d7005304",
      "installation_id": "81F4B083-DC04-4047-8304-2969997400A0",
      "time": 1678895687765,
      "ios_app_state": {
        "network_addresses": [
          "fe80::aede:48ff:fe00:1122",
          "172.16.100.10",
          "fe80::e850:c6ff:fe8e:f01b",
          "fe80::e850:c6ff:fe8e:f01b",
          "fe80::296b:e36f:9541:34a2",
          "fe80::3c9e:888c:92a7:a2fc",
          "fe80::ce81:b1c:bd2c:69e",
          "10.8.0.94"
        ],
        "window_root_view_controller_titles": [
          "RNSScreen"
        ],
        "sdk_version": "v2.1.1"
      }
    },
    {
      "user_id": "5744b8888b3b8145d7005304",
      "installation_id": "81F4B083-DC04-4047-8304-2969997400A0",
      "time": 1678895807977,
      "ios_app_state": {
        "network_addresses": [
          "fe80::aede:48ff:fe00:1122",
          "172.16.100.10",
          "fe80::e850:c6ff:fe8e:f01b",
          "fe80::e850:c6ff:fe8e:f01b",
          "fe80::296b:e36f:9541:34a2",
          "fe80::3c9e:888c:92a7:a2fc",
          "fe80::ce81:b1c:bd2c:69e",
          "10.8.0.94"
        ],
        "sdk_version": "v2.1.1"
      }
    }
  ],
  "beacon_key": "ab6ecc6664"
}


```

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Dev changes

- Send data to Sift for fraud prevention - mdole

</details>

[PRO-87]: https://artsyproduct.atlassian.net/browse/PRO-87?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/store/AuthModel.ts [hunk 1]
```diff
@@ -909,6 +910,8 @@ export const getAuthModel = (): AuthModel => ({
       }
     }
 
+    SiftReactNative.unsetUserId()
```
Comment (mdole) [inline src/app/store/AuthModel.ts:913]: I tried to test this but it's not even clear from their code what it does or why it's necessary :/

**Comment by gkartalis** [comment]
Let me know if you need any help with setting up android, and also for the tests that fail, I think we just need to mock the `SiftReactNative` module (you can see how we can do that [here](https://www.reactnative.guide/7-testing/7.4-mocking-rn-modules.html))

**Comment by mdole** [comment]
Thanks @gkartalis ! Added a mock for the tests & I was able to get Android set up and am seeing some events but also some weirdness. Going to mark this PR as draft while I investigate 🙏

File: src/setupJest.ts [hunk 2]
```diff
@@ -108,6 +108,10 @@ jest.mock("tipsi-stripe", () => ({
   createTokenWithCard: jest.fn(),
 }))
 
+jest.mock("sift-react-native", () => ({
```
Comment (gkartalis) [inline src/setupJest.ts:111]: 🔥

**Review by rajsam003 — APPROVED** [review]
LGTM

**Review by iskounen — COMMENTED** [review]
LGTM so far 👍

**Comment by mdole** [comment]
oookay so long story short, Sift's wrapper currently does _not_ correctly upload events on Android devices. This is a bug with their wrapper, and I've flagged with their engineering team, but they're unwilling to give us a timeline on fixing it. More info in [Slack](https://artsy.slack.com/archives/C04TGU3EW9L/p1679585929504219?thread_ts=1678989492.181899&cid=C04TGU3EW9L). It sends basic information when a call to `upload()` is made, including as part of `setSiftConfig()`, but it does not automatically collect, batch, and send events over time like it's supposed to.

We hemmed and hawed a bit, and I started going down the path of implementing their native SDK for Android, before we decided that since more of our traffic is iOS-based (and iOS is working with the RN wrapper), we'd be OK with shipping with minimal events on Android and fully-functional events on iOS, which this provides. And hopefully Sift will ship a fix sooner rather than later and we can put this whole thing behind us.

@iskounen this is mergeable now!

**Pull request merged**